content/blog/prototyping/index.md

```diff
@@ -0,0 +1,767 @@
+
+## Why People Think Rust Is Not Good For Prototyping
+
+The common narrative goes like this:
+
+> When you start writing a program, you don't know what you want and you change your mind pretty often.
+> Rust pushes back when you change your mind because the type system is very strict.
+> On top of that, getting your idea to compile takes longer than in other languages, so the feedback loop is slower.
+
+That's why people believe that Rust isn't a good fit for prototyping.
+
+I've found that developers from other languages and Rust beginners often share this preconception about Rust.
+These developers stumble over the strict type system and the borrow checker while trying to sketch out a solution.
+They believe that with Rust you're either at 0% or 100% done (everything works and has no undefined behavior) and there's nothing in between.
+
+Here are some typical misbeliefs: 
+
+1. "Memory safety and prototyping just don't go together."
+2. "Ownership and borrowing take out the fun of prototyping."
+3. "You have to get all the details right from the beginning."
+4. "Rust always requires you to handle errors."
+
+These are all [common misconceptions](https://medium.com/@victor.ronin/love-hate-relationship-with-rust-language-part-2-c36f57d5485d) and they are not true. 
+
+**It turns out you can avoid all of these pitfalls and still get a lot of value from prototyping in Rust.**
+
+## What Makes Rust Great for Prototyping?
+
+### Solid Foundation From The Beginning
+
+In an ideal world, you'd have plenty of time to perfect your code, but in reality, deadlines are tight and you have to make compromises.
+
+Eventually management will say "ship it" and you'll have to live with the code you wrote.
+If that's the case, a rewrite is often not an option.
+
+Rust allows you to have a solid foundation from the start.
+Even the first version is often good enough for production.
+
+### Strong Safety Guarantees 
+
+Rust allows you to avoid many pitfalls by being very explicit about error conditions, but also by providing escape hatches in case you choose to ignore them for a while.
+This saves you a lot of time hunting down bugs later on.
+
+### Single Language for Everything
+
+To me, it's super nice to have a single language I know well and can use for all stages of a project. 
+A language that I can learn and understand all the way down to the implementation of the standard library itself (simply by reading to the source code).
+Other languages are often written in C/C++, and it's hard to understand what's going on under the hood.
+
+### Great Teaching Tool
+
+I prototype in Rust frequently when I need to explain systems-level concepts to clients or sketch out a prototype for a new project of my own.
+It's a great way to test my assumptions before I invest too much time in a flawed design. 
+
+## Problems with Prototyping in Other Languages
```

Comment:
I have similar doubts about this section, as I have about the previous one. Some notes:

- It seems like it should come before the previous one, maybe? Showing me why Python is not so perfect for prototyping is a perfect setup for why I should consider Rust.
- You're spending a lot of words here on Python before getting into the core of the article, how to use Rust for prototyping.
- Maybe condense this down, merge it with the condensed version of the previous section, and leave any details for an appendix or follow-up article?